src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Float16.java

```diff
@@ -0,0 +1,1667 @@
+        // round up (in magnitude) is a function of the least
+        // significant bit (LSB), the next bit position (the round
+        // position), and the sticky bit (whether there are any
+        // nonzero bits in the exact result to the right of the round
+        // digit). An increment occurs in three cases:
+        //
+        // LSB  Round Sticky
+        // 0    1     1
+        // 1    1     0
+        // 1    1     1
+        // See "Computer Arithmetic Algorithms," Koren, Table 4.9
+
+        long lsb    = f_signif_bits & (1L << 42 + expdelta);
+        long round  = f_signif_bits & (1L << 41 + expdelta);
+        long sticky = f_signif_bits & ((1L << 41 + expdelta) - 1);
+
+        if (round != 0 && ((lsb | sticky) != 0 )) {
+            signif_bits++;
+        }
+
+        // No bits set in significand beyond the *first* exponent bit,
+        // not just the significand; quantity is added to the exponent
+        // to implement a carry out from rounding the significand.
+        assert (0xf800 & signif_bits) == 0x0;
+
+        return new Float16((short)(sign_bit | ( ((exp + 15) << 10) + signif_bits ) ));
+    }
+
+    /**
+     * Returns a {@code Float16} holding the floating-point value
+     * represented by the argument string.
+     *
+     * The grammar of strings accepted by this method is the same as
+     * that accepted by {@link Double#valueOf(String)}. The rounding
+     * policy is also analogous to the one used by that method, a
+     * valid input is regarded as an exact numerical value that is
+     * rounded once to the nearest representable {@code Float16} value.
+     *
+     * @apiNote
+     * This method corresponds to the convertFromDecimalCharacter and
+     * convertFromHexCharacter operations defined in IEEE 754.
+     *
+     * @param  s the string to be parsed.
+     * @return the {@code Float16} value represented by the string
+     *         argument.
+     * @throws NullPointerException  if the string is null
+     * @throws NumberFormatException if the string does not contain a
+     *               parsable {@code Float16}.
+     * @see    java.lang.Float#valueOf(String)
+     */
+    public static Float16 valueOf(String s) throws NumberFormatException {
+        s = s.trim(); // Legacy behavior from analagous methods on
+                      // Float and Double.
+
+        // Trial conversion from String -> double. Do quick range
+        // check for a pass-through, then check for possibility of
+        // double-rounding and another conversion using
+        // BigInteger/BigDecimal, if needed.
+        double trialResult = Double.parseDouble(s);
+        // After this point, the trimmed string is known to be
+        // syntactically well-formed; should be able to operate on
+        // characters rather than codepoints.
+
+        if (trialResult == 0.0 // handles signed zeros
+            || Math.abs(trialResult) > (65504.0 + 32.0) || // Float.MAX_VALUE + ulp(MAX_VALUE)
```

Comment:
```suggestion
            || Math.abs(trialResult) > (65504.0 + 16.0) || // Float.MAX_VALUE + ulp(MAX_VALUE) / 2
```

Reply:
When writing the code, I didn't think through all the possible boundary conditions near Float16.MAX_VALUE + 0.5*ulp(Float16.MAX_VALUE) so I bumped out the overflow limit.